ros2param/test/test_verb_dump.py

```diff
@@ -12,75 +12,112 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from io import StringIO
+import contextlib
 import os
+import sys
 import tempfile
-import threading
 import time
 import unittest
-from unittest.mock import patch
 import xmlrpc
 
+from launch import LaunchDescription
+from launch.actions import ExecuteProcess
+from launch_ros.actions import Node
+import launch_testing
+import launch_testing.actions
+import launch_testing.asserts
+import launch_testing.markers
+import launch_testing.tools
+
+import pytest
+
 import rclpy
-from rclpy.executors import MultiThreadedExecutor
-from rclpy.parameter import PARAMETER_SEPARATOR_STRING
 
-from ros2cli import cli
 from ros2cli.node.strategy import NodeStrategy
 
 TEST_NODE = 'test_node'
 TEST_NAMESPACE = '/foo'
 
-TEST_TIMEOUT = 5.0
-
-EXPECTED_PARAMETER_FILE = """\
-test_node:
-  ros__parameters:
-    bool_param: true
-    double_param: 1.23
-    foo:
-      bar:
-        str_param: foobar
-      str_param: foo
-    int_param: 42
-    str_param: Hello World
-    use_sim_time: false
-"""
+TEST_TIMEOUT = 20.0
+
+EXPECTED_PARAMETER_FILE = (
+    f'{TEST_NODE}:\n'
+    '  ros__parameters:\n'
+    '    bool_param: true\n'
+    '    double_param: 1.23\n'
+    '    foo:\n'
+    '      bar:\n'
+    '        str_param: foobar\n'
+    '      str_param: foo\n'
+    '    int_param: 42\n'
+    '    str_param: Hello World\n'
+    '    use_sim_time: false\n'
+)
+
+# TODO(jacobperron): Enable testing with all RMW implementations when they pass
+# @launch_testing.parametrize('rmw_implementation', get_available_rmw_implementations())
+@pytest.mark.rostest
+@launch_testing.parametrize('rmw_implementation', ['rmw_fastrtps_cpp'])
```

Comment:
What's the failure mode if Fast-RTPS is not part of the build? Does it make more sense to parameterize now and remove cyclonedds?

Reply:
Good point. We can just remove the parameterization altogether for now.

Reply:
To answer your question, I'd expect the test to fail because of the unknown RMW implementation.

Reply:
Actually, with #482 resolved, it looks like the test should work with all of our existing RMWs. It was easier to parameterize now than not (56e34da)